server/src/main/java/org/opensearch/cluster/service/ClusterManagerTaskThrottler.java

```diff
@@ -32,15 +32,15 @@
  */
 public class ClusterManagerTaskThrottler implements TaskBatcherListener {
     private static final Logger logger = LogManager.getLogger(ClusterManagerTaskThrottler.class);
-    private static final String TASK_SUFFIX = "-task";
+    public static final ThrottlingKey DEFAULT_THROTTLING_KEY = new ThrottlingKey("default-task-key", false);
 
     public static final Setting<Settings> THRESHOLD_SETTINGS = Setting.groupSetting(
         "cluster_manager.throttling.thresholds.",
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
 
-    protected Map<String, ClusterManagerThrottlingKey> THROTTLING_TASK_KEYS = new ConcurrentHashMap<>();
+    protected Map<String, ThrottlingKey> THROTTLING_TASK_KEYS = new ConcurrentHashMap<>();
```

Comment:
Maybe we just need a Set<ThrottlingKey>

Reply:
We also want to perform get as well, when user set the threshold against key. 

That's why I have kept it as map, with set every time we will need to iterate over all the entries of set to verify if the tasks is on boarded for throttling or not.

Reply:
Can't we do Set#contains and override hash code/equals accordingly?

Reply:
Sure, but for validating setting we need to do get as well to verify is throttling is enabled (true/false flag) for that key or not. For every threshold setting update we will need to iterate over the set.